src/Resources/PageResource.php

```diff
@@ -372,4 +387,29 @@ protected static function afterValidation(NovaRequest $request, $validator)
                 ->add('slug', __('pages::pages.validation.uniqueSlug'));
         }
     }
+
+    protected function getTemplateDependentFields(): array
+    {
+        return collect($this->getPageTemplates())
+            ->filter(
+                fn(string $templateName) => method_exists(
+                    $this,
+                    "fieldsFor" . ucfirst($templateName)
+                )
+            )
+            ->map(function (string $templateName) {
+                $method = "fieldsFor" . ucfirst($templateName);
```

Comment:
Gaat dit goed voor template namen met een streepje erin?

Reply:
Jammer dat die `ucfirst` redundant is. Ik dacht dat die wel in een `map` voor `filter` zou kunnen, maar je hebt natuurlijk `$templateName` nodig voor de `dependsOn`.

Reply:
Nee, uiteraard heeft Laravel daar iets mooiers voor. Ik heb `Str::studly` ervoor in de plaats gezet.

En die redundantie is inderdaad jammer. Ik heb getwijfeld of ik de `filter` en `map` samen zou voegen tot een `reduce`, maar ik denk dat dit makkelijker te lezen is.